src/Microsoft.Performance.SDK.Runtime/Extensibility/DataExtensions/DataCookers/CompositeDataCookerReference.cs

```diff
@@ -1,19 +1,89 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-using Microsoft.Performance.SDK.Extensibility;
-using Microsoft.Performance.SDK.Extensibility.DataCooking;
 using System;
 using System.Diagnostics;
+using Microsoft.Performance.SDK.Extensibility;
+using Microsoft.Performance.SDK.Extensibility.DataCooking;
 
 namespace Microsoft.Performance.SDK.Runtime.Extensibility.DataExtensions.DataCookers
 {
+    /// <summary>
+    ///     Represents a reference to a composite data cooker.
+    /// </summary>
     internal class CompositeDataCookerReference
```

Comment:
Would you mind sealing this class? If you're ok with that, then 'instanceLock' could be made private (it probably should have been already).